src/deadline/blender_adaptor/BlenderClient/render_handlers/default_blender_handler.py

```
@@ -65,28 +64,20 @@ def _ensure_camera(self, data: dict) -> str:
         _logger.debug(f"Fetched camera to render: {camera}")
         return camera
 
-    def _get_layer_to_render(self, data: dict) -> Optional[str]:
-        """Gets the view_layer to render from the provided data.
+    def _ensure_view_layer(self) -> None:
+        """
+        Ensure that the view layer is properly set before rendering.
 
-        If no view_layer is specified in `data`, return None.
+        Raises:
+            RuntimeError: If no view layer is set on the adaptor
         """
-        target = data.get("view_layer")
-        if not target:
-            _logger.debug("No view_layer specified in data.")
-            return None
 
-        # In Blender, set the right scene.
-        scene = bpy.data.scenes[self.scene_name]
-        bpy.context.window.scene = scene
+        if self.view_layer_name is None:
+            raise RuntimeError("View layer was not provided in init data")
```

Nit, this error message is not entirely accurate 
```suggestion
            raise RuntimeError("ensure_view_layer must not be called before set_view_layer")
```

---

Actually I think we might not want to raise an exception at all since `view_layer` is not required in `init-data` or `run-data`

---

In this case we can likely default to whichever layer(s) are set to enabled in the scene, which I think was the previous behavior

---

Sure! Testing it out quickly, passing an empty string for `layer` in the `render` call will by default render every layer in the scene at once, i.e. what the current adaptor does.

As-is this would only be available for people using custom job templates with the adaptor, but if we really find it's necessary we can add the option in the submitter later

---

Nice, do we still need to get the name of the layer from the scene so that the places we reference `view_layer_name` are properly updated (mainly output paths)?

---

I'm a bit hesitant because if a scene has multiple layers, we'd just be arbitrarily picking one. Old submitter versions will still write layer names, so we should be fine!